Sources/StreamChatSwiftUI/ChatChannel/MessageList/MessageViewModel.swift

```diff
@@ -0,0 +1,152 @@
+//
+// Copyright © 2025 Stream.io Inc. All rights reserved.
+//
+
+import Combine
+import StreamChat
+
+/// The view model that contains the logic for displaying a message in the message list view.
+open class MessageViewModel: ObservableObject {
+    @Injected(\.utils) private var utils
+    @Injected(\.chatClient) private var chatClient
+
+    public private(set) var message: ChatMessage
+    public private(set) var channel: ChatChannel?
+    private var originalTranslationsStore: MessageOriginalTranslationsStore
+    private var cancellables = Set<AnyCancellable>()
+
+    public init(
+        message: ChatMessage,
+        channel: ChatChannel?,
+        originalTranslationsStore: MessageOriginalTranslationsStore
+    ) {
+        self.message = message
+        self.channel = channel
+        self.originalTranslationsStore = originalTranslationsStore
+        self.originalTranslationsStore.$originalTextMessageIds.sink(
+            receiveValue: { [weak self] _ in
+                self?.objectWillChange.send()
+            }
+        )
+        .store(in: &cancellables)
```

Comment:
Since this is a nested observable object, we need to notify the parent object that the translations have changed

Reply:
why do you need this? Can't you just make the store `@Published`?

Reply:
No it wont work, because this is nested observable object, even with @Published

Reply:
is that still an issue? It's there from iOS 13 😅

Reply:
Yeah, I think only new @Observable macro fixes this